### PR TITLE
feat: add validator identity info

### PR DESCRIPTION
**Pull request by jstarry**
- add validator identity info to cluster info cache
- create validator identity component
- add validator identity info to all of the validator screens (list, overview, tds, detail)
- add avatar image

Screenshot 2019-07-17 21 52 46
Screenshot 2019-07-17 21 52 39
Screenshot 2019-07-18 15 24 16

### CODE REVIEW

**Comment by jstarry** [comment]
This depends on a currently unreleased version of `@solana/web3.js`, after a new version with this change https://github.com/solana-labs/solana-web3.js/pull/407 is published, should be good to go.

**Comment by sunnygleason** [comment]
@jstarry this is looking pretty great, thank you! I just had a couple suggestions, let me know what you think, I'm ready to merge once they're all set!

**Comment by CriesofCarrots** [comment]
@sunnygleason , did you maybe start a review that you didn't complete? I'm curious to see your suggestions

**Comment by sunnygleason** [comment]
@CriesofCarrots @jstarry review submitted - sorry about that, someday I'll figure this github thing out! 😹

File: src/App.js [hunk 1]
```diff
@@ -141,11 +141,14 @@ class App extends Component {
   parseClusterInfo(data) {
     let voting = data.voting;
     let gossip = data.cluster;
+    let validators = data.validators;
```
[the remark this answers is not in the document]
Comment (jstarry) [reply src/App.js]: how about `identities`?

File: api/api.js [hunk 2]
```diff
@@ -550,4 +565,69 @@ app.get('/cluster-info', (req, res) => {
...
+          }
+          info.pubkey = validatorKeyStr;
+          return info;
+        }
+      }
+    }),
+  );
+
+  const infoList = results.filter(r => r);
+  const keybaseUsernames = infoList.map(info => info.keybaseUsername).filter(u => u).join(',');
```
[the remark this answers is not in the document]
Comment (jstarry) [reply api/api.js]: good point, I don't see it documented anywhere but tested it myself and maxed out at 50. Will do some batching!

File: api/api.js [hunk 3]
```diff
@@ -550,4 +565,69 @@ app.get('/cluster-info', (req, res) => {
...
+  const avatarMap = new Map();
+  if (keybaseUsernames.length > 0) {
+    const keybaseApiUrl = `https://keybase.io/_/api/1.0/user/lookup.json?usernames=${keybaseUsernames}&fields=pictures,basics`;
+    const keybaseResponse = await fetch(keybaseApiUrl);
```
[the remark this answers is not in the document]
Comment (jstarry) [reply api/api.js]: Ah yeah this definitely needs a try/catch

**Comment by sunnygleason** [comment]
@jstarry this is great, thank you!
@mvines merging FYI, let me know if anything pops up on "edge"
@CriesofCarrots thank you too!

**Pull request merged**